name.abuchen.portfolio.tests/src/name/abuchen/portfolio/snapshot/security/DividendCalculationTest.java

```diff
@@ -0,0 +1,203 @@
+        transactions.add(createDividendTransaction(LocalDateTime.of(2018, 04, 15, 12, 00)));
+        transactions.add(createDividendTransaction(LocalDateTime.of(2018, 07, 15, 12, 00)));
+        transactions.add(createDividendTransaction(LocalDateTime.of(2018, 10, 15, 12, 00)));
+
+        DividendCalculation dividends = Calculation.perform(DividendCalculation.class, converter, security,
+                        transactions);
+
+        assertEquals(Periodicity.QUARTERLY, dividends.getPeriodicity());
+    }
+
+    @Test
+    public void periodicitySemiAnnualTest()
+    {
+
+        List<Transaction> transactions = new ArrayList<>();
+        transactions.add(createDividendTransaction(LocalDateTime.of(2019, 01, 15, 12, 00)));
+        transactions.add(createDividendTransaction(LocalDateTime.of(2019, 07, 15, 12, 00)));
+
+        DividendCalculation dividends = Calculation.perform(DividendCalculation.class, converter, security,
+                        transactions);
+
+        assertEquals(Periodicity.SEMIANNUAL, dividends.getPeriodicity());
+    }
+
+    @Test
+    public void periodicityQuarterlyTest()
+    {
+
+        List<Transaction> transactions = new ArrayList<>();
+        transactions.add(createDividendTransaction(LocalDateTime.of(2019, 01, 15, 12, 00)));
+        transactions.add(createDividendTransaction(LocalDateTime.of(2019, 04, 15, 12, 00)));
+
+        DividendCalculation dividends = Calculation.perform(DividendCalculation.class, converter, security,
+                        transactions);
+
+        assertEquals(Periodicity.QUARTERLY, dividends.getPeriodicity());
+    }
+
+    @Test
+    public void periodicityMonthlyTest()
+    {
+
+        List<Transaction> transactions = new ArrayList<>();
+        transactions.add(createDividendTransaction(LocalDateTime.of(2019, 01, 15, 12, 00)));
+        transactions.add(createDividendTransaction(LocalDateTime.of(2019, 02, 15, 12, 00)));
+
+        DividendCalculation dividends = Calculation.perform(DividendCalculation.class, converter, security,
+                        transactions);
+
+        assertEquals(Periodicity.MONTHLY, dividends.getPeriodicity());
+    }
+
+    @Test
+    public void rateOfReturnCalculationTest()
+    {
+
+        List<Transaction> transactions = new ArrayList<>();
+        // We buy some shares, 1000$, 10 Shares -> 100/Share
+        transactions.add(new PortfolioTransaction(LocalDateTime.of(2019, 01, 14, 12, 00), security.getCurrencyCode(),
+                        1000L, security, 10L, Type.BUY, 0L, 0L));
+        transactions.add(createDividendTransaction(LocalDateTime.of(2019, 01, 15, 12, 00)));
+        transactions.add(createDividendTransaction(LocalDateTime.of(2020, 01, 15, 12, 00)));
+
+        // We need to calculate the costs, in order to get the average return
+        CostCalculation cost = Calculation.perform(CostCalculation.class, converter, security, transactions);
```

Comment:
The cost Object is never used -> not needed

Reply:
Jein... 

lass mal den Test laufen ohne das die Kosten berechnet wurden. Dann failt der Test. 



Er kann dann den average return nicht berechnen... ich verstehe zu wenig wie die Kalkulation genau läuft, evtl kannst du dir das kurz anschauen?